tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/remoting/ReconnectTest.java

```diff
@@ -330,9 +330,13 @@ public void testReattachTimeout() throws Exception {
       server.start();
       // imitate session reattach timeout
       Interceptor reattachInterceptor = new Interceptor() {
+
+         boolean reattached;
+
          @Override
          public boolean intercept(Packet packet, RemotingConnection connection) throws ActiveMQException {
-            if (packet.getType() == PacketImpl.REATTACH_SESSION) {
+            if (!reattached && packet.getType() == PacketImpl.REATTACH_SESSION) {
```

Comment:
@clebertsuconic re-attach is meant to occur just the once, so making the imitation done via interceptor do that.